heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraController.java

```diff
@@ -11,144 +11,21 @@
-    if (containerId != -1) {
-      auroraCmd.add(String.format("%s/%s", jobSpec, Integer.toString(containerId)));
-    } else {
-      auroraCmd.add(jobSpec);
-    }
-
-    appendAuroraCommandOptions(auroraCmd, isVerbose);
-
-    return runProcess(auroraCmd);
-  }
-
-  void removeContainers(Set<PackingPlan.ContainerPlan> containersToRemove) {
-    String instancesToKill = getInstancesIdsToKill(containersToRemove);
-    //aurora job kill <cluster>/<role>/<env>/<name>/<instance_ids>
-    List<String> auroraCmd = new ArrayList<>(Arrays.asList(
-        "aurora", "job", "kill", jobSpec + "/" + instancesToKill));
-    LOG.info(String.format(
-        "Killing %s aurora containers: %s", containersToRemove.size(), auroraCmd));
-    if (!runProcess(auroraCmd)) {
-      throw new RuntimeException("Failed to kill freed aurora instances: " + instancesToKill);
-    }
-  }
-
-  void addContainers(Integer count) {
-    //aurora job add <cluster>/<role>/<env>/<name>/<instance_id> <count>
-    //clone instance 0
-    List<String> auroraCmd = new ArrayList<>(Arrays.asList(
-        "aurora", "job", "add", "--wait-until", "RUNNING", jobSpec + "/0", count.toString()));
-    LOG.info(String.format("Requesting %s new aurora containers %s", count, auroraCmd));
-    if (!runProcess(auroraCmd)) {
-      throw new RuntimeException("Failed to create " + count + " new aurora instances");
-    }
-  }
-
-  // Utils method for unit tests
-  @VisibleForTesting
-  boolean runProcess(List<String> auroraCmd) {
-    StringBuilder stdout = new StringBuilder();
-    StringBuilder stderr = new StringBuilder();
-    int status =
-        ShellUtils.runProcess(auroraCmd.toArray(new String[auroraCmd.size()]), stdout, stderr);
-
-    if (status != 0) {
-      LOG.severe(String.format(
-          "Failed to run process. Command=%s, STDOUT=%s, STDERR=%s", auroraCmd, stdout, stderr));
-    }
-    return status == 0;
-  }
-
-  private static String getInstancesIdsToKill(Set<PackingPlan.ContainerPlan> containersToRemove) {
-    StringBuilder ids = new StringBuilder();
-    for (PackingPlan.ContainerPlan containerPlan : containersToRemove) {
-      if (ids.length() > 0) {
-        ids.append(",");
-      }
-      ids.append(containerPlan.getId());
-    }
-    return ids.toString();
-  }
-
-  // Static method to append verbose and batching options if needed
-  private static void appendAuroraCommandOptions(List<String> auroraCmd, boolean isVerbose) {
-    // Append verbose if needed
-    if (isVerbose) {
-      auroraCmd.add("--verbose");
-    }
+public interface AuroraController {
 
-    // Append batch size.
-    // Note that we can not use "--no-batching" since "restart" command does not accept it.
-    // So we play a small trick here by setting batch size Integer.MAX_VALUE.
-    auroraCmd.add("--batch-size");
-    auroraCmd.add(Integer.toString(Integer.MAX_VALUE));
-  }
+  boolean createJob(String auroraFilename, Map<String, String> bindings);
```

Comment:
Good catch. I refactored `createJob` to remove the `auroraFilename`, which only applies to the CLI impl. I also renamed `bindings` to `auroraProperties` and changed the key from a string to an enum to make it's usage clear.